java/gazelle/private/types/types.go

```diff
@@ -0,0 +1,85 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+	"unicode"
+)
+
+type PackageName struct {
+	Name string
+}
+
+func NewPackageName(name string) PackageName {
+	return PackageName{
+		Name: name,
+	}
+}
+
+func PackageNameLess(l, r PackageName) bool {
+	return l.Name < r.Name
+}
+
+type ClassName struct {
+	packageName        PackageName
+	bareOuterClassName string
+	innerClassNames    []string
```

Comment:
Classes can be nested to an arbitrary depth. I can't tell if `bareOuterClassName` is the containing class or the top-level class